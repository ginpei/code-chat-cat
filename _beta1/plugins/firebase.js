import firebase from 'firebase';
import 'firebase/storage';
import config from './firebase.config.js';

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

export default firebase;
