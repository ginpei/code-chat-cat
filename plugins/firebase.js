import firebase from 'firebase'

if (!firebase.apps.length) {
  firebase.initializeApp({
    // apiKey: 'AIzaSyCchW1XVLnIfG_wre1PKync7ZVfUylGRV4',
    // authDomain: 'hello-nuxt-firebase.firebaseapp.com',
    databaseURL: 'https://hello-nuxt-firebase.firebaseio.com/',
    // projectId: 'hello-nuxt-firebase',
    // storageBucket: process.env.STORAGEBUCKET,
  });
}

export default firebase;
