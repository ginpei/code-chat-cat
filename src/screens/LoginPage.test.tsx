import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import firebase from '../middleware/firebase';
import ConnectedLoginPage from './LoginPage';

const LoginPage = ConnectedLoginPage.WrappedComponent;

describe('<LoginPage>', () => {
  describe('while logged in', () => {
    it('just show a message', () => {
      const wrapper = shallow((
        <LoginPage
          loggedIn={true}
        />
      ));
      expect(wrapper.text()).toMatch('Logged in');
    });
  });

  describe('while logged out', () => {
    let wrapper: ShallowWrapper;
    let auth: jest.SpyInstance;

    beforeEach(() => {
      auth = jest.spyOn(firebase, 'auth')
        .mockReturnValue({} as any);
      Object.assign(auth, {
        EmailAuthProvider: {
          PROVIDER_ID: 'EmailAuthProvider',
        },
        GithubAuthProvider: {
          PROVIDER_ID: 'GithubAuthProvider',
        },
      });

      wrapper = shallow((
        <LoginPage
          loggedIn={false}
        />
      ));
    });

    afterEach(() => {
      auth.mockRestore();
    });

    it('show the login form', () => {
      expect(wrapper.text()).toMatch('Login');

      // how to find the component StyledFirebaseAuth?
      // expect(wrapper.find('StyledFirebaseAuth')).toHaveLength(1);
      // expect(wrapper.find('#firebaseui_container')).toHaveLength(1);
    });

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
