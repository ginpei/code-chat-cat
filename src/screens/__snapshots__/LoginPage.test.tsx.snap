// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage> while logged out matches snapshot 1`] = `ShallowWrapper {}`;
